### ISSUE TITLE
api: rename `runTemplateRef` to `runTemplateName`

### ISSUE THREAD

**Issue by cirocosta** [vmware-tanzu/cartographer#255]
### Description of problem

_**tl;dr: inconsistency with core resources.**_

those who might already be used to looking at fields that have `*Ref`  might expect that `runTemplateRef` is namespace-scoped reference (i.e., one that allows _name_ **and** _namespace_), like, `pipelineRef` in the case of a PipelineRun.

transitioning RunTemplate to ClusterRunTemplate we went from having `runTemplateRef` properly allowing you to specify name _and_ namespace to just a name (as it's now cluster-wide), but didn't change the `Ref` there in the name of the field to indicate the lack of namespace (i.e., we didn't change to `runTemplateName`, which might not be very consistent).

### Proposed solution

ps.: I'm not entirely sure if we even should do this or not (please let me know what you think!)

```gherkin
Given I want to point at a cluster-wide runtemplate
When I discover the `spec.runTemplateRef` field
Then I don't get confused about the possibility of having a namespace specified in that object
```

I believe that the way to go here is to mimic what, e.g, statefulsets do when it comes to pointing at a cluster-wide resource storageclass: use `name` (string field), like, `storageClassName` (i.e., we could go with `runTemteName`).

### Example

```yaml
kind: ClusterRunTemplate
metadata:
  name: foo

---

kind: Pipeline
spec:
 runTemplateName: foo
```

**Comment by waciumawanjohi** [comment]
I hear where you are coming from. At the same time, I think it is quite likely that we will introduce a namespaced scoped RunTemplate at some point. At which point we'll need the `namespace` field and we'd have to change back to the current `runTemplateRef`.

**Comment by waciumawanjohi** [comment]
On a related note, `runTemplateRef` is of type `TemplateReference` [link](https://github.com/vmware-tanzu/cartographer/blob/c0737a94b3a3c35a8d87c9e4c078476de1d580e6/pkg/apis/v1alpha1/pipeline.go#L58), which accepts a Kind [link](https://github.com/vmware-tanzu/cartographer/blob/c0737a94b3a3c35a8d87c9e4c078476de1d580e6/pkg/apis/v1alpha1/pipeline.go#L74). Right now we enforce that this entry is either `ClusterRunTemplate` or empty [link](https://github.com/vmware-tanzu/cartographer/blob/f1964bb0ce2cdf85f7eb3bee80d130e89197da37/pkg/repository/repository.go#L166-L171) and [link](https://github.com/vmware-tanzu/cartographer/blob/f1964bb0ce2cdf85f7eb3bee80d130e89197da37/pkg/registrar/map_functions.go#L141). We should remove `kind` from `TemplateReference`.

**Comment by squeedee** [comment]
> We should remove kind from TemplateReference

Agreed

> At which point we'll need the namespace field and we'd have to change back to the current runTemplateRef

Let's defer until this is asked for. It'll be a breaking change, but so will what happens with all templates anyway.

**Comment by scothis** [comment]
Even if you only need the name field. I'd recommend using a [LocalObjectReference](https://pkg.go.dev/k8s.io/api/core/v1#LocalObjectReference)-like struct. The key advantage is that it's more clearly a reference and it's much easier to introduce new fields as the api evolves in a backwards compatible way.

**Comment by martyspiewak** [comment]
@scothis So are you advocating for keeping `runTemplateRef` (and just removing `kind`)?

**Comment by squeedee** [comment]
Wont fix for 0.1.x as it introduces a breaking change to customers. At least one customer has a runTemplateRef.Kind.
